dotnet/src/SemanticKernel/CoreSkills/ConditionalSkill.cs

```diff
@@ -0,0 +1,316 @@
+            EvaluateConditionPrompt,
+            skillName: nameof(ConditionalSkill),
+            description: "Evaluate a condition group and returns TRUE or FALSE",
+            maxTokens: 100,
+            temperature: 0,
+            topP: 0.5);
+
+        this._evaluateIfBranchFunction = kernel.CreateSemanticFunction(
+            ExtractThenOrElseFromIfPrompt,
+            skillName: nameof(ConditionalSkill),
+            description: "Extract the content of the first child tag from the root If element",
+            maxTokens: 1000,
+            temperature: 0,
+            topP: 0.5);
+
+        if (completionBackend is not null)
+        {
+            this._ifStructureCheckFunction.SetAIBackend(() => completionBackend);
+            this._evaluateIfBranchFunction.SetAIBackend(() => completionBackend);
+            this._evaluateConditionFunction.SetAIBackend(() => completionBackend);
+        }
+    }
+
+    /// <summary>
+    /// Get a planner if statement content and output then or else contents depending on the conditional evaluation.
+    /// </summary>
+    /// <param name="ifContent">If statement content.</param>
+    /// <param name="context"> The context to use </param>
+    /// <returns>Then or Else contents depending on the conditional evaluation</returns>
+    /// <remarks>
+    /// This skill is initially intended to be used only by the Plan Runner.
+    /// </remarks>
+    [SKFunction("Get a planner if statement content and output then or else contents depending on the conditional evaluation.")]
+    [SKFunctionName("If")]
+    public async Task<SKContext> IfAsync(string ifContent, SKContext context)
+    {
+        var usedVariables = await this.GetVariablesAndEnsureIfStructureIsValidAsync(ifContent, context).ConfigureAwait(false);
+
+        bool conditionEvaluation = await this.EvaluateConditionAsync(ifContent, usedVariables, context).ConfigureAwait(false);
+
+        return await this.GetThenOrElseBranchAsync(ifContent, conditionEvaluation, context).ConfigureAwait(false);
+    }
+
+    private async Task<SKContext> GetThenOrElseBranchAsync(string ifContent, bool trueCondition, SKContext context)
+    {
+        var branchVariables = new ContextVariables(ifContent);
+        context.Variables.Set("EvaluateIfBranchTag", trueCondition ? "Then" : "Else");
+        context.Variables.Set("IfStatementContent", ifContent);
+
+        return (await this._evaluateIfBranchFunction.InvokeAsync(context).ConfigureAwait(false));
+    }
+
+    private async Task<IEnumerable<string>> GetVariablesAndEnsureIfStructureIsValidAsync(string statement, SKContext context)
+    {
+        context.Variables.Set("IfStatementContent", statement);
+        var llmCheckFunctionResponse = (await this._ifStructureCheckFunction.InvokeAsync(statement, context).ConfigureAwait(false)).ToString();
+
+        JsonNode llmResponse = this.GetValuesFromResponse(llmCheckFunctionResponse);
+        var valid = llmResponse["valid"]!.GetValue<bool>();
```

Comment:
Are we sure key `valid` exists in `llmResponse`?

Reply:
Yep, in the prompt I enforce the return of the valid property. Run it many iterations and got the valid there.